clang/lib/3C/DeclRewriter.cpp

```diff
@@ -735,11 +735,14 @@ FunctionDeclBuilder::buildDeclVar(PVConstraint *IntCV, PVConstraint *ExtCV,
   // just fall back to reconstructing the declaration from the PVConstraint.
   ParmVarDecl *PVD = dyn_cast<ParmVarDecl>(Decl);
   if (PVD) {
-    Type = getSourceText(PVD->getSourceRange(), *Context);
-    if (!Type.empty()) {
-      // Great, we got the original source including any itype and bounds.
-      IType = "";
-      return;
+    SourceRange Range = PVD->getSourceRange();
+    if (Range.isValid()) {
```

Comment:
Hm, I thought that `getSourceText` could only return non-empty if `isValid` returned true (that's why I removed the `isValid` check), but I must have been wrong about that.